### ISSUE TITLE
How should we preprocess data that works with both Catboost and other algorithm

### ISSUE THREAD

**Issue by herbzhao** [catboost/catboost#360]
Problem: {How can we preprocess data that works with both Catboost and other algorithm}

Catboost is great at doing the job alone, but to be careful, I always want to try multiple algorithm (sklearn for example) for the same problem. 

I wonder what's the recommended way of pre-process the data? It  #seems like catboost also transform catgorical data into numerical ones, is it possible to retain the transformed dataset and feed into other algorithms?

**Comment by Noxoomo** [comment]
Hi!

It's not possible currently.
Plus CatBoost generates new numerical features during learning via greedy algorithm, not only on pre-process stage

**Comment by herbzhao** [comment]
Thanks! 
In terms of preprocessing that enables both catboost and sklearn. Is label encoding, one hot encoding or feature hashing gonna do any harm for the performance of catboost?

**Comment by Noxoomo** [comment]
One-hot encoding will harm performance (learning time will be significantly slower)

Hashing could reduce models quality, because you'll lose some information, which CatBoost could utilize

**Comment by hugoncosta** [comment]
@Noxoomo what would you advice to use instead for text features?

**Comment by annaveronika** [comment]
@hugoncosta For text features it is usually beneficial to build some embeddings and then calculate distances and pass them as numeric features. For example, you can cluster them and pass distances to clusters. You can also use one-hot encoding, we have not implemented the fast support for sparse data. We are working on that though. If you want to do one-hot encoding for catboost, you need to use one_hot_max_size parameter.

As for categorical features - we do calculate things similar to one-hot encoding, but we calculate them on several permutations plus we do greedy feature combinations on several permutations also. You cannot do it as preprocessing, you need to do this inside the algorithm.

So as for this issue, I don't think we have much to do here, so I'm closing it. If you have more questions or suggestions, feel free to open a new one.
